src/main/java/xyz/kmahyyg/eshopdemo/controller/GoodController.java

```diff
@@ -0,0 +1,60 @@
+package xyz.kmahyyg.eshopdemo.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import xyz.kmahyyg.eshopdemo.dao.SysItemsDao;
+import xyz.kmahyyg.eshopdemo.dao.SysUsersDao;
+import xyz.kmahyyg.eshopdemo.model.SysItems;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Controller
+public class GoodController {
+    @Autowired
+    private SysItemsDao sysItemsDao;
+
+    @RequestMapping("show/good/detail")
+    public String getGoodDetail(ModelMap map,Integer itemId){
+        //map.addAttribute("host", itemId);
+        //System.out.println("id is:"+itemId);
+        SysItems showItem = sysItemsDao.selectById(itemId);
+        if(showItem != null){
+            map.addAttribute("itemName", showItem.getName());
+            map.addAttribute("itemPrice", showItem.getPrice());
+            map.addAttribute("itemStatus", showItem.getStatus());
+            map.addAttribute("itemImage", showItem.getImage());
+            map.addAttribute("itemDesc", showItem.getDescr());
```

Comment:
Two questions:

1. How about we just build another model to wrap this `showItem` for convenient code reuse?
2. Please keep in mind that any item is added by a untrusted user who owned a shop on our platform.
Since processing input while inserting an item into database might cause unintended data loss or misunderstanding. I think we should just escape all data we get from user at show time.

I've already implemented an example for escaping and sanitize user input. Please check https://github.com/kmahyyg/shit-ssm-eshop/blob/master/src/main/java/xyz/kmahyyg/eshopdemo/utils/UserInputSanitizer.java for more details. As for this example, I use ESAPI from OWASP.

Reply:
Also, we use Thymeleaf as our template engine, which might introduce SSTI vulnerability. Please do check and escape user input to prevent this from happening further.